webapp/src/widget/todo_toast/todo_toast.tsx

```diff
@@ -0,0 +1,79 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+import React, {useEffect, useCallback} from 'react';
+
+import './todo_toast.scss';
+import {CSSTransition} from 'react-transition-group';
+
+import {generateClassName} from '../../utils';
+import CompassIcon from '../../components/icons/compassIcons';
+import IconButton from '../iconButton/iconButton';
+
+type Props = {
+    close: () => void,
+    children?: React.ReactNode,
+    submit: PropTypes.func.isRequired,
+    title?: string,
+    content: {
+        icon: string,
+        message: string,
+        undo: () => void,
+    },
+    className?: string,
+}
+
+function TodoToast(props: Props): JSX.Element {
+    const {close, content} = props;
+
+    const closeToast = useCallback(close, [close]);
+    const undoTodo = useCallback(() => {
+        content.undo();
+        close();
+    }, [content.undo(), close]);
```

Comment:
```suggestion
    }, [content.undo, close]);
```
You want to make sure the function is the same function, not the result of the function (which probably is void).